src/lib.rs

```diff
@@ -28,100 +28,75 @@ pub extern "C" fn wapc_init() {
 
 fn validate_envvar_with_rule(rule: &Rule, env_vars: &[apicore::EnvVar]) -> Result<()> {
     let error_message: String;
-    let resource_env_var: HashSet<settings::EnvVar> = env_vars
-        .iter()
-        .map(|envvar| settings::EnvVar {
-            name: envvar.name.clone(),
-            value: envvar.value.clone(),
-        })
-        .collect();
-    let resource_env_var_names: HashSet<settings::EnvVar> = env_vars
-        .iter()
-        .map(|envvar| settings::EnvVar {
-            name: envvar.name.clone(),
-            value: None,
-        })
-        .collect();
-    let validation_envvar_with_values: HashSet<settings::EnvVar> = rule
-        .environment_variables
-        .clone()
-        .into_iter()
-        .filter(|envvar| envvar.value.is_some())
-        .collect();
-    let validation_envvar_name_only: HashSet<settings::EnvVar> = rule
-        .environment_variables
-        .clone()
-        .into_iter()
-        .filter(|envvar| envvar.value.is_none())
-        .collect();
+    let resource_env_var_names: HashSet<String> =
+        env_vars.iter().map(|envvar| envvar.name.clone()).collect();
+
     match rule.reject {
```

Comment:
I find this `match` to be hard to digest. I propose the following refactoring:

- Create a new file `src/operators.rs`
- Inside of the file define functions that take as input two `HashSet` and return `Result<()>`
- Do table testing, like you're doing, but at a function level. This makes everything more understandable
- Refactor this `match` to invoke the appropriate function